bless_client/bless_client.go

```
@@ -0,0 +1,76 @@
+package main
+
+import (
+    "os"
+    "fmt"
+    "io/ioutil"
+    "encoding/json"
+    "github.com/aws/aws-sdk-go/aws"
+    "github.com/aws/aws-sdk-go/aws/session"
+    "github.com/aws/aws-sdk-go/service/lambda"
+    "reflect"
+)
+
+func check(e error) {
+    if e != nil {
+        panic(e)
+    }
+}
+
+func main() {
+    args := os.Args[1:]
+    if len(args) != 9 {
+        fmt.Println(
+            `Usage: bless_client.py region lambda_function_name bastion_user bastion_user_ip
+             remote_username bastion_ip bastion_command <id_rsa.pub to sign>
+             <output id_rsa-cert.pub>`)
+        os.Exit(1)
+    }
+
+    region, lambda_function_name, bastion_user, bastion_user_ip, remote_username, bastion_ip, bastion_command, public_key_filename, certificate_filename := args[0], args[1], args[2], args[3], args[4], args[5], args[6], args[7], args[8]
+
+    fmt.Println(region, lambda_function_name, bastion_user, bastion_user_ip, remote_username, bastion_ip, bastion_command, public_key_filename, certificate_filename)
+
+    public_key, err := ioutil.ReadFile(public_key_filename)
+    check(err)
+    fmt.Print(string(public_key))
+    fmt.Println(reflect.TypeOf(public_key))
+
+    payload := map[string]string{"bastion_user": bastion_user, "bastion_user_ip": bastion_user_ip,
+                                 "remote_username": remote_username, "bastion_ip": bastion_ip,
+                                 "command": bastion_command, "public_key_to_sign": string(public_key)}
+    payload_json, _ := json.Marshal(payload)
```

do not ignore error. do something like payload_json, err :=
and then check if err != nil